CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SwiftSyntaxCompatibilityFile.swift

```diff
@@ -0,0 +1,135 @@
+                \(raw: child.varName) = value
+              }
+            }
+            """
+          )
+        }
+      }
+
+      let deprecatedNames = layoutNode.children
+        .filter { $0.isUnexpectedNodes }
+        .compactMap { $0.deprecatedName?.withFirstCharacterLowercased }
+        .joined(separator: ", ")
+
+      try! InitializerDeclSyntax(
+        """
+        @available(*, deprecated, message: "Use an initializer with a \(raw: deprecatedNames) argument(s).")
+        @_disfavoredOverload
+        \(layoutNode.generateInitializerDeclHeader(useDeprecatedChildName: true))
+        """
+      ) {
+        FunctionCallExprSyntax(callee: ExprSyntax("self.init")) {
+          TupleExprElementSyntax(label: "leadingTrivia", expression: ExprSyntax("leadingTrivia"))
+          for child in layoutNode.children {
+            if child.isUnexpectedNodes {
+              TupleExprElementSyntax(expression: ExprSyntax("\(raw: child.deprecatedVarName)"))
+            } else {
+              TupleExprElementSyntax(label: "\(child.varName)", expression: ExprSyntax("\(raw: child.deprecatedVarName)"))
+            }
+          }
+          TupleExprElementSyntax(label: "trailingTrivia", expression: ExprSyntax("trailingTrivia"))
+        }
+      }
+    }
+  }
+
+  try ExtensionDeclSyntax("public extension SyntaxProtocol") {
+    DeclSyntax(
+      """
+      @available(*, deprecated, message: "Use detached computed property instead.")
+      func detach() -> Self {
+        return detached
+      }
+      """
+    )
+  }
+
+  try ExtensionDeclSyntax("public extension TokenKind") {
+    DeclSyntax(
+      """
+      @available(*, deprecated, renamed: "leftSquare")
+      static var leftSquareBracket: TokenKind {
+        return .leftSquare
+      }
+      """
+    )
+
+    DeclSyntax(
+      """
+      @available(*, deprecated, renamed: "rightSquare")
+      static var rightSquareBracket: TokenKind {
+        return .rightSquare
+      }
+      """
+    )
+  }
+
+  try ExtensionDeclSyntax("public extension TokenSyntax") {
+    DeclSyntax(
+      """
+      @available(*, deprecated, renamed: "leftSquareToken")
+      static func leftSquareBracketToken(
+        leadingTrivia: Trivia = [],
+        trailingTrivia: Trivia = [],
+        presence: SourcePresence = .present
+      ) -> TokenSyntax {
+        return .leftSquareToken(
+          leadingTrivia: leadingTrivia,
+          trailingTrivia: trailingTrivia,
+          presence: presence
+        )
+      }
+      """
+    )
+
+    DeclSyntax(
+      """
+      @available(*, deprecated, renamed: "rightSquareToken")
+      static func rightSquareBracketToken(
+        leadingTrivia: Trivia = [],
+        trailingTrivia: Trivia = [],
+        presence: SourcePresence = .present
+      ) -> TokenSyntax {
+        return .rightSquareToken(
+          leadingTrivia: leadingTrivia,
+          trailingTrivia: trailingTrivia,
+          presence: presence
+        )
+      }
+      """
+    )
+  }
```

Comment:
I would prefer to not generate these static accessors. We can just leave them in a non-generated SwiftSyntaxCompatibility.swift file and name this file RenamedChildrenCompatibility.swift or something similar.